Llama Frontend Created

Saupernova13 opened the pull request:
I have developed a program that enables users to interact with the Llama 3 Ai, and implemented the extra feature of a chat history selector. My solution was done in C# as a WPF Desktop Application.

Saupernova13 committed:
Commit of Initial Prototype

A basic prototype designed to take in text based user input and prompt Llama 3 for a response.

The desired features to add are:

- Chat History/Multiple Chat Instances

Saupernova13 committed:
Chat History Implemented

- New Chat Button Added
- Chat History Menu Added
- Ability to Load Past Chats Added

Saupernova13 committed:
Merge branch 'chat-history-implementation'

Saupernova13 committed:
Updated Main GUI

- Improved Visual Aesthetic of Main GUI

Saupernova13 committed:
Updated InputBox UI

- Updated InputBox UI

Saupernova13 committed:
Clear Chat Button Added

- Clear All Chat History
- Added Ability To Submit Prompt With Enter Key
- Slight UI Modifications

Saupernova13 committed:
Demo Video Added

- Added Demo Video

leeroya commented on aiChatterBox/packages/Newtonsoft.Json.13.0.3/Newtonsoft.Json.13.0.3.nupkg:
Have a look at maybe adding packages, to the gitignore. Most build systems have the ability to pull public packages from nuget.org so the need have binary files in source control are not needed.

leeroya commented on aiChatterBox/aiChatterBox/bin/Debug/Newtonsoft.Json.dll:
Consider adding the bin folder in your .gitignore so that temporary compile files are not tracked in the source control system, they are not required and can be reproduced in a build system or on a contributors machine.

leeroya commented on aiChatterBox/aiChatterBox/obj/Debug/MainWindow.g.cs:
Consider adding the obj folder into your .gitignore file for the same reason as mentioned for the bin folder

leeroya commented on Sauraav_Jayrajh_Derivco_MagicSpoon_Demo.mp4:
Thank you for the video, shows that you went the extra mile. Consider creating a docs folder in the root of the repository and have your documentation and videos there, with a comment in the root readme.md file that points to it.

You can achieve this by making use of links in markdown [For more information...](docs/readme.md)

leeroya commented on .gitignore:
Here is a nice resource to use for your project .gitignore file: [gitignore](https://github.com/github/gitignore/tree/main)

leeroya commented on aiChatterBox/aiChatterBox/MainWindow.xaml.cs:
```diff
@@ -0,0 +1,284 @@
+
+        //Reuseable Method to Handle Submission of Prompt
+        public async void submitPrompt()
+        {
```
I am a little rusty on WinForms and WPF, which is called UWP application and MAUI, however maybe consider haveing a look at this video [Stop using async void in C#!](https://www.youtube.com/watch?v=ZFWxSQ-KjUc) and see if it applies. Maybe try use:

```
public async Task submitPrompt()
{
}
```
As void returns nothing and Task will return task which allows threads to safely return

leeroya commented on aiChatterBox/aiChatterBox/MainWindow.xaml.cs:
```diff
@@ -0,0 +1,284 @@
+            }
+
+            //Add User Input to Chat
+            chats[currentChatIndex].Add($"ME: {inputPrompt}");
```
I understand why you doing this:
`chats[currentChatIndex]`

However would it not be easier to manage a private variable that holds that object and type so that it is easier to read and support, functionally the same.

```
private int currentChatIndex = -1;
private ControlType _chat;
private List<List<string>> chats = new List<List<string>>();


private void listView_PastChats_SelectionChanged(object sender, SelectionChangedEventArgs e)
{
}
```

leeroya commented on aiChatterBox/aiChatterBox/MainWindow.xaml.cs:
```diff
@@ -0,0 +1,284 @@
+            };
+
+            //Format and Send Request
+            var json = JsonConvert.SerializeObject(requestBody);
```
Consider using System.Text instead of: using Newtonsoft.Json; as its a good idea to limit the amount of dependencies where possible. .Net has matured their built in support for JSON which covers 90% of operations.

leeroya commented on aiChatterBox/aiChatterBox/MainWindow.xaml.cs:
```diff
@@ -0,0 +1,284 @@
+        //Client to Handle API Interactions
+        private static readonly HttpClient client = new HttpClient();
+        //Adress of Ollama Running On Machine, Change if Needed
+        public static string localhost = "http://localhost:11434";
```
Consider having a way to pull from a configuration file, then have an control or form that allows the user to update this value.

leeroya reviewed (changes requested):
# Summary

Your application effectively meets its stated goals and functions well. While there are minor areas for improvement, these insights often come from working with various code bases, which you'll gain more experience with over time.

Consider setting up a GitHub Action to package your application and include some developer requirements in the README.md.

I personally use VSCode, and when I cloned your repository, the initial dotnet build ...sln command didn't work. However, using msbuild /t:some arguments was successful, and Visual Studio also worked. To enhance usability, you might want to create a script, such as a build.ps1, to streamline the compile step. Many open-source and business repositories benefit from having a straightforward script for testing newly cloned repositories.

Saupernova13 committed:
Delete aiChatterBox/packages directory

Saupernova13 committed:
Delete aiChatterBox/aiChatterBox/.vs directory

Saupernova13 committed:
Delete aiChatterBox/aiChatterBox/bin/Debug directory

Saupernova13 committed:
Delete aiChatterBox/aiChatterBox/obj/Debug directory

Saupernova13 committed:
Added Config Loading/Saving

- Added Ability to Save a Host Config
- Added Ability to Load a Saved Host Config
- Created Docs Folder
- Added Relevant Directories to .gitignore
- Changes submitPrompt Method to Task

Saupernova13 committed:
Migration to System.Text.Json

- Migrated from Newtonsoft.Json to System.Text.Json

Saupernova13 committed:
Typo Fix

- Fixed Typo

Saupernova13 committed:
Better Chat Handling

- Introduced Private Variable to Handle Chats Instead Of Directly Calling Index
- Allows For User To Override Default Config Value

Saupernova13 committed:
New Demo Video Added

- New Demo Video Added Showcasing New Feature

Saupernova13 committed:
Updated Readme

- Updated Readme

Saupernova13 committed:
Thumbnails Added

- Thumbnails Added For Videos

Saupernova13 committed:
Update README.md

Saupernova13 commented:
Chanes have been made and will be updated via new Pull Request
